pact/coin-contract/coin.pact

```diff
@@ -176,64 +171,51 @@
         })
       ))
 
-  (defun delete-coin (delete-account create-chain-id create-account create-account-guard quantity)
-    @doc "Burn QUANTITY-many coins for DELETE-ACCOUNT on the current chain, and \
-         \produce an SPV receipt which may be manually redeemed for an SPV      \
-         \proof. Once a proof is obtained, the user may call 'create-coin' and  \
-         \consume the proof on CREATE-CHAIN-ID, crediting CREATE-ACCOUNT        \
-         \QUANTITY-many coins."
-
-    @model [(property (> amount 0.0))]
-
-    (with-capability (TRANSFER)
-      (debit delete-account quantity)
-
-      { "create-chain-id": create-chain-id
-      , "create-account": create-account
-      , "create-account-guard": create-account-guard
-      , "quantity": quantity
-      , "delete-block-height": (at 'block-height (chain-data))
-      , "delete-chain-id": (at 'chain-id (chain-data))
-      , "delete-account": delete-account
-      , "delete-tx-hash": (tx-hash)
-      })
-    )
-
-  (defun create-coin (proof)
-    @doc "Consume an SPV proof for a number of coins, and credit the account   \
-         \associated with the proof the quantify of coins burned on the source \
-         \chain by the burn account. Note: must be called on the correct chain \
-         \id as specified in the proof."
-
-    (let ((outputs (verify-spv "TXOUT" proof)))
-      (bind outputs
-        { "create-chain-id":= create-chain-id
-        , "create-account" := create-account
+  (defpact cross-chain-transfer
+    ( delete-account:string
+      create-chain-id:string
+      create-account:string
+      create-account-guard:guard
+      quantity:decimal )
+
+    @doc "Step 1: Burn QUANTITY-many coins for DELETE-ACCOUNT on the current chain, \
+         \and produce an SPV receipt which may be manually redeemed for an SPV      \
+         \proof. Once a proof is obtained, the user may call 'create-coin' and      \
+         \consume the proof on CREATE-CHAIN-ID, crediting CREATE-ACCOUNT QUANTITY-  \
+         \many coins.                                                               \
+         \                                                                          \
+         \Step 2: Consume an SPV proof for a number of coins, and credit the        \
+         \account associated with the proof the quantify of coins burned on the     \
+         \source chain by the burn account. Note: must be called on the correct     \
+         \chain id as specified in the proof."
+
+    @model [ (property (> quantity 0.0))
+           , (property (not (= create-chain-id "")))
+           ]
+
+    (step
+      (with-capability (TRANSFER)
+        (debit delete-account quantity)
```

Comment:
I think we should consider that calling `cross-chain-transfer` with the current chain ID is a bug and should be enforced as such

Reply:
I've added the `enforce` and updated the docs